addons/google_recaptcha/static/src/js/signup.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+import { ReCaptcha } from "@google_recaptcha/js/recaptcha";
+
+const CaptchaFunctionality = {
+    events: {
+        submit: "_onSubmit",
+    },
+
+    init() {
+        this._super(...arguments);
+        this._recaptcha = new ReCaptcha();
+    },
+
+    async willStart() {
+        return this._recaptcha.loadLibs();
+    },
+
+    _onSubmit(event) {
+        if (!this.$el.find("input[name='recaptcha_token_response']").length) {
+            event.preventDefault();
```

Comment:
Normally I consider naming the "event" variable `event` a bad practice. If you omit the definition of the "event" variable, this code will still work... on some browsers, including Chrome. Meaning that if you have a habit of naming your event variables `event`, you will create bugs on some browsers at some point when you'll forgot to name the parameter explicitly (and see your code working on your own browser).

I am always naming them `ev`, that way if I forgot to define the parameter, then it just crashes, on all browsers.